tools/Utilities/StatusCheckLogger.cs

```diff
@@ -0,0 +1,146 @@
+﻿using Octokit;
+
+namespace Utilities;
+
+/// <summary>
+/// Record for a single diagnostic
+/// </summary>
+/// <param name="file">The source file in the PR</param>
+/// <param name="Message">The message for the output daignostic</param>
+/// <param name="Id">The error message ID</param>
+/// <param name="StartLine">The start line (index from 1)</param>
+/// <param name="EndLine">The end line (index from 1)</param>
+public record Diagnostic(string file, int StartLine, int EndLine, string Message, string Id);
+
+/// <summary>
+/// This class writes the status of the check to the console in the format GitHub supports
+/// </summary>
+/// <remarks>
+/// For all of our tools, if all error and warning messages are formatted correctly, GitHub
+/// will show those errors and warnings inline in the files tab for the PR. Let's format
+/// them correctly.
+/// </remarks>
+/// <param name="pathToRoot">The path to the root of the repository</param>
+/// <param name="toolName">The name of the tool that is running the check</param>
+public class StatusCheckLogger(string pathToRoot, string toolName)
+{
+    private List<NewCheckRunAnnotation> annotations = [];
+    public bool Success { get; private set; } = true;
+
+    // Utility method to format the path to unix style, from the root of the repository.
+    private string FormatPath(string path) => Path.GetRelativePath(pathToRoot, path).Replace(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
+
+    private void WriteMessageToConsole(string prefix, Diagnostic d) => Console.WriteLine($"{prefix}{toolName}-{d.Id}::file={FormatPath(d.file)},line={d.StartLine}::{d.Message}");
+
+    /// <summary>
+    /// Log a notice from the status check
+    /// </summary>
+    /// <param name="d">The diagnostic</param>
+    /// <remarks>
+    /// Add the diagnostic to the annotation list and
+    /// log the diagnostic information to console.
+    /// </remarks>
+    public void LogNotice(Diagnostic d)
+    {
+        WriteMessageToConsole("", d);
+        annotations.Add(
+            new(FormatPath(d.file),
+            d.StartLine, d.EndLine,
+            CheckAnnotationLevel.Notice, $"{d.Id}::{d.Message}")
+        );
+    }
+
+    /// <summary>
+    /// Log a warning from the status check
+    /// </summary>
+    /// <param name="d">The diagnostic</param>
+    /// <remarks>
+    /// Add the diagnostic to the annotation list and
+    /// log the warning notice to the console.
+    /// </remarks>
+    public void LogWarning(Diagnostic d)
+    {
+        WriteMessageToConsole("⚠️", d);
+        annotations.Add(
+            new(FormatPath(d.file),
+            d.StartLine, d.EndLine,
+            CheckAnnotationLevel.Warning, $"{d.Id}::{d.Message}")
+        );
+        Success = false;
```

Comment:
This looks like it's treating warnings as errors... I don't know whether we want that.